modules/zip/doc_classes/ZIPReader.xml

```diff
@@ -14,6 +15,32 @@
 		    var res := reader.read_file("hello.txt")
 		    reader.close()
 		    return res
+
+		# Extract all files from a ZIP archive, preserving the directories within.
+		# This acts like the "Extract all" functionality from most archive managers.
+		func extract_all_from_zip():
+		    var reader := ZIPReader.new()
+		    reader.open("res://archive.zip")
+
+		    # Destination directory for the extracted files (this folder must exist before extraction).
+		    # Not all ZIP archives put everything in a single root folder,
+		    # which means several files/folders may be created in `root_dir` after extraction.
+		    var root_dir := DirAccess.open("user://")
```

Comment:
```suggestion
		    var root_dir := DirAccess.open("res://")
```
Is there any consideration for extracting the files into `res://` instead? It would make it a bit more immediately obvious that the sample code works when it's copied. But I understand if it's bad practice to recommend that for other reasons.

Reply:
It's only useful for that purpose in the editor, you can't write to "res://" in exported projects, so this script would have to be `@tool` or just run in the editor, I don't think it's a good idea to write to "res" with this

Reply:
Indeed, writing to `user://` is preferable as it showcases good practices here.